syncano/connection.py

```diff
@@ -290,6 +290,8 @@ def get_response_content(self, url, response):
 
         # Validation error
         if is_client_error(response.status_code):
+            if response.status_code == 400 and 'Revision mismatch.' in content.get('non_field_errors', [''])[0]:
```

Comment:
In theory, it may not necessarily be at [0] position as there may be more non_field_errors. Considering a custom exception in backend for this though.